Mathlib/NumberTheory/LSeries/HurwitzZetaOdd.lean

```diff
@@ -0,0 +1,592 @@
+analytic continuation is not at all obvious); we in fact construct them as Mellin transforms of
+various versions of the Jacobi theta function.
+
+## Main definitions and theorems
+
+* `completedHurwitzZetaOdd`: the completed Hurwitz zeta function
+* `completedSinZeta`: the completed cosine zeta function
+* `differentiable_completedHurwitzZetaOdd` and `differentiable_completedSinZeta`:
+  differentiability on `ℂ`
+* `completedHurwitzZetaOdd_one_sub`: the functional equation
+  `completedHurwitzZetaOdd a (1 - s) = completedSinZeta a s`
+* `hasSum_int_hurwitzZetaOdd` and `hasSum_nat_sinZeta`: relation between
+  the zeta functions and corresponding Dirichlet series for `1 < re s`
+-/
+
+noncomputable section
+
+open Complex hiding abs_of_nonneg
+open Filter Topology Asymptotics Real Set MeasureTheory
+open scoped ComplexConjugate
+
+section kernel_defs
+/-!
+## Definitions and elementary properties of kernels
+-/
+
+/-- Variant of `jacobiTheta₂'` which we introduce to simplify some formulae. -/
+def jacobiTheta₂'' (z τ : ℂ) : ℂ :=
+  cexp (π * I * z ^ 2 * τ) * (jacobiTheta₂' (z * τ) τ / (2 * π * I) + z * jacobiTheta₂ (z * τ) τ)
+
+lemma jacobiTheta₂''_conj (z τ : ℂ) :
+    conj (jacobiTheta₂'' z τ) = jacobiTheta₂'' (conj z) (-conj τ) := by
+  simp only [jacobiTheta₂'', jacobiTheta₂'_conj, jacobiTheta₂_conj,
+    ← exp_conj, conj_ofReal, conj_I, map_mul, map_add, map_div₀, mul_neg, map_pow, map_ofNat,
+    neg_mul, div_neg, neg_div, jacobiTheta₂'_neg_left, jacobiTheta₂_neg_left]
+
+/-- Restatement of `jacobiTheta₂'_add_left'`: the function `jacobiTheta₂''` is 1-periodic in `z`. -/
+lemma jacobiTheta₂''_add_left (z τ : ℂ) : jacobiTheta₂'' (z + 1) τ = jacobiTheta₂'' z τ := by
+  simp only [jacobiTheta₂'', add_mul z 1, one_mul, jacobiTheta₂'_add_left', jacobiTheta₂_add_left']
+  generalize jacobiTheta₂ (z * τ) τ = J
+  generalize jacobiTheta₂' (z * τ) τ = J'
+  -- clear denominator
+  simp_rw [div_add' _ _ _ two_pi_I_ne_zero,  ← mul_div_assoc]
+  refine congr_arg (· / (2 * π * I)) ?_
+  -- get all exponential terms to left
+  rw [← mul_assoc _ (cexp _), mul_comm z (cexp _), mul_assoc (cexp _), ← mul_add,
+    mul_assoc (cexp _), ← mul_add, ← mul_assoc (cexp _), ← Complex.exp_add]
+  congrm (cexp ?_ * ?_) <;> ring
+
+lemma jacobiTheta₂''_neg_left (z τ : ℂ) : jacobiTheta₂'' (-z) τ = -jacobiTheta₂'' z τ := by
+  simp only [jacobiTheta₂'', jacobiTheta₂'_neg_left, jacobiTheta₂_neg_left,
+    neg_mul, neg_div, ← neg_add, mul_neg, neg_sq]
+
+lemma jacobiTheta₂'_functional_equation' (z τ : ℂ) :
+    jacobiTheta₂' z τ = (-2 * π) / (-I * τ) ^ (3 / 2 : ℂ) * jacobiTheta₂'' z (-1 / τ) := by
+  rcases eq_or_ne τ 0 with rfl | hτ
+  · rw [jacobiTheta₂'_undef _ (by simp), mul_zero, zero_cpow (by norm_num), div_zero, zero_mul]
+  have aux1 : (-2 * π : ℂ) / (2 * π * I) = I := by rw [div_eq_iff two_pi_I_ne_zero, mul_comm I,
+    mul_assoc _ I I, I_mul_I, neg_mul, mul_neg, mul_one]
```

Comment:
```suggestion
  have aux1 : (-2 * π : ℂ) / (2 * π * I) = I := by
    rw [div_eq_iff two_pi_I_ne_zero, mul_comm I, mul_assoc _ I I, I_mul_I, neg_mul, mul_neg,
      mul_one]
```
I think the proof should start on the same line only if it finishes on that line.